lib/matplotlib/backends/backend_pgf.py

```diff
@@ -322,46 +328,33 @@ def finalize_latex(latex):
 
     def get_width_height_descent(self, text, prop):
         """
-        Get the width, total height and descent for a text typeset by the
-        current LaTeX environment.
+        Get the width, total height, and descent (in TeX points) for a text
+        typeset by the current LaTeX environment.
         """
+        return self._get_box_metrics(_escape_and_apply_props(text, prop))
 
-        # apply font properties and define textbox
-        prop_cmds = _font_properties_str(prop)
-        textbox = "\\sbox0{%s %s}" % (prop_cmds, text)
-
-        # check cache
-        if textbox in self.str_cache:
-            return self.str_cache[textbox]
-
-        # send textbox to LaTeX and wait for prompt
-        self._stdin_writeln(textbox)
-        try:
-            self._expect_prompt()
-        except LatexError as e:
-            raise ValueError("Error processing '{}'\nLaTeX Output:\n{}"
-                             .format(text, e.latex_output)) from e
-
-        # typeout width, height and text offset of the last textbox
-        self._stdin_writeln(r"\typeout{\the\wd0,\the\ht0,\the\dp0}")
-        # read answer from latex and advance to the next prompt
+    def _get_box_metrics(self, tex):
```

Comment:
Should probably get a note that `__init__` wraps this with a per-instance-cache.

Reply:
done